packages/astro/src/core/build/index.ts

```diff
@@ -198,7 +198,9 @@ class AstroBuilder {
 		await runHookBuildDone({
 			config: this.settings.config,
 			pages: pageNames,
-			routes: Object.values(allPages).map((pd) => pd.route),
+			routes: Object.values(allPages)
+				.flat()
+				.map((pd) => pd.route),
```

Comment:
This is totally unrelated, but `pd` is a slur in French, also not sure what it stands for, should probably be a more descriptive name.